animation_nodes/nodes/mesh/mesh_points_scatter.py

```diff
@@ -29,31 +38,46 @@ def create(self):
         self.newInput("Integer", "Amount", "amount", value = 10, minValue = 0)
         self.newInput("Float List", "Weights", "weights", hide = True)
 
-        self.newOutput("Vector List", "Points", "points")
+        self.newOutput("Matrix List", "Matrices", "matrices")
+        self.newOutput("Vector List", "Vectors", "vectors")
 
     def draw(self, layout):
         row = layout.row(align = True)
         row.prop(self, "nodeSeed", text = "Node Seed")
+        layout.prop(self, "mode", text = "")
 
     def drawAdvanced(self, layout):
         layout.prop(self, "methodType", text = "Use Advanced Method for Mesh sampling")
 
-    def execute(self, mesh, seed, amount, weights):
+    def getExecutionCode(self, required):
+        yield "matrices = self.execute_RandomPointsScatter(mesh, seed, amount, weights)"
+
+        if "vectors" in required:
+            yield "vectors = AN.nodes.matrix.c_utils.extractMatrixTranslations(matrices)"
+
+    def execute_RandomPointsScatter(self, mesh, seed, amount, weights):
         vertices = mesh.vertices
         polygons = mesh.polygons
+        edges = mesh.edges
 
-        if len(vertices) == 0 or len(polygons) == 0 or amount == 0:
-            return Vector3DList()
-
-        if polygons.polyLengths.getMaxValue() > 3:
-            if self.methodType:
-                polygons = triangulatePolygonsUsingEarClipMethod(vertices, polygons)
-            else:
-                polygons = triangulatePolygonsUsingFanSpanMethod(polygons)
+        if len(vertices) == 0 or amount == 0:
+            return Matrix4x4List()
 
         weights = VirtualDoubleList.create(weights, 1)
         seed = cantorPair(int(max(seed, 0)), self.nodeSeed)
-        return randomPointsScatter(vertices, polygons, weights, seed, max(amount, 0))
+        if self.mode == "POLYGONS":
+            if len(polygons) == 0: return Matrix4x4List()
+            if polygons.polyLengths.getMaxValue() > 3:
+                if self.methodType:
+                    polygons = triangulatePolygonsUsingEarClipMethod(vertices, polygons)
+                else:
+                    polygons = triangulatePolygonsUsingFanSpanMethod(polygons)
+            polyNormals = calculatePolygonNormals(vertices, polygons)
+            return randomPointsScatter(vertices, polygons, polyNormals, weights, seed, max(amount, 0))
```

Comment:
To unify things, rename the function to `scatterPointsOnPolygons` and rename the other function to `scatterPointsOnEdges`.

Reply:
Done.